x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_management/hooks/use_prebuilt_rules_upgrade.tsx

```diff
@@ -111,6 +141,9 @@ export function usePrebuiltRulesUpgrade({
         )
       );
 
+      const { noConflictRuleIds, solvableConflictRuleIds, nonSolvableConflictRuleIds } =
+        getConflictRuleIds(ruleIds);
+
```

Comment:
This can probably be combined with the logic in `conflictRuleIdsSet` but this allows us to easily use the same logic in the `ALL_RULES` function for the confirmation modal